examples/gno.land/r/demo/gnofundme/gnofundme.gno

```diff
@@ -0,0 +1,253 @@
+package gnofundme
+
+import (
+	"bytes"
+	"std"
+	"strconv"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/avl"
+	pkg "gno.land/p/demo/gnofundme"
+	"gno.land/p/demo/ufmt"
+)
+
+var (
+	campaigns      *avl.Tree
+	users          *avl.Tree
+	campaignPrefix = "/r/demo/gnofundme:"
+)
+
+func init() {
+	campaigns = avl.NewTree()
+	users = avl.NewTree()
+}
+
+func NewUser(name string, email string) string {
+	// get the user address
+	address := std.GetOrigCaller()
+	// create the user
+	user := pkg.NewUser(name, email, address)
+
+	// the avl.tree key will be address to simplify the fetching
+	users.Set(address.String(), user)
+
+	return user.GetName() + " added successfully"
+}
+
+func NewCampaign(title string, description string, goal uint, beginT int64, deadlineT int64, beneficiary std.Address) (int, string) {
+	// get the caller as the owner address
+	caller := std.GetOrigCaller()
+	user := pkg.NewUser("user1", "email1", caller)
```

Comment:
What if the caller has already created his user? Shouldn't you get the information from the `avl.Tree` called users?

Reply:
that's what I did in the beginning, but when I worked on the front, I wanted to gain time and make some tests so I modified the code, that's why you see verification (users.Get) after that.